### DISCUSSION TITLE
[Discussion] How to query the decompressed metadata from znode ?

### DISCUSSION THREAD

**Discussion by kaori-seasons** [apache/pulsar#19668] [General]
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### Motivation

Hi community. I'm targeting [PIP-123](https://github.com/apache/pulsar/issues/13346) interest. I spent two weeks at work doing some tuning research. 

### Solution

The mechanism I understand that needs to be modified is this: call the java program in the zookeeper-shell to pass in some command line parameters, and then put it into the managed-ledger cursor through the zk client server such as curator,
However, since the managed-ledger here may touch multiple copies of the bk node metadata when reading and writing zk,
Now there are options such as rocksdb, so it needs to be unified and compatible in the underlying logic of AdminCli.

Pulsar's internal protocol serialization data (xxxxmessage.proto) in the znode needs to download metadata without breaking the existing internal internal protocol. How should I do? Because in my opinion need to do a layer of decompression abstraction

### Alternatives

none

### Anything else?

none

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

**Comment by tisonkun** [comment]
I don't know what's the actual problem here. But for reading encoded data on znode, you need to decode it in the CLI class and that's all.
